server/fleet/windows_mdm.go

```diff
@@ -45,6 +45,10 @@ type MDMWindowsConfigProfile struct {
 //
 // Returns an error if these conditions are not met.
 func (m *MDMWindowsConfigProfile) ValidateUserProvided() error {
+	if _, ok := microsoft_mdm.FleetReservedProfileNames()[m.Name]; ok {
+		return fmt.Errorf("Profile name %q is not allowed.", m.Name)
```

Comment:
@marko-lisica or @noahtalerman  you probably want to review the copy of this new error case : to set the Windows OS Updates settings, we create a new Fleet-controlled profile. As such, this ensures custom profiles cannot be created with that name. The message was adusted from the similar message when a custom profile is validated against Fleet-reserved Apple Payload Identifiers.

Reply:
@mna I guess `%q` is a variable that will display "Disk encryption"?

> The message was adusted from the similar message when a custom profile is validated against Fleet-reserved Apple Payload Identifiers.

Makes sense to me to have this consistent with macOS.

Reply:
Yeah sorry, forgot to mention, the only reserved name for Windows profiles currently is "Windows OS Updates".

Reply:
Great, looks good to me.